app/tests/spec/models/auth_brokers/web-channel.js

```diff
@@ -177,28 +177,170 @@ function (chai, sinon, WebChannelAuthenticationBroker, Relier, User, FxaClientWr
       });
     });
 
+    describe('beforeSignUpConfirmationPoll', function () {
+      it('does not persist key-fetch material by default', function () {
+        setupCompletesOAuthTest();
+        assert.isFalse(broker.relier.wantsKeys());
+        account.set('keyFetchToken', 'keyFetchToken');
+        account.set('unwrapBKey', 'unwrapBKey');
+
+        return broker.persist()
+          .then(function () {
+            return broker.beforeSignUpConfirmationPoll(account);
+          })
+          .then(function () {
+            assert.isUndefined(broker.session.oauth.keyFetchToken);
+            assert.isUndefined(broker.session.oauth.unwrapBKey);
+          });
+      });
+
+      it('persists key-fetch material if the relier wants keys', function () {
+        setupCompletesOAuthTest();
+        sinon.stub(broker.relier, 'wantsKeys', function () {
+          return true;
+        });
+
+        return broker.persist()
+          .then(function () {
+            account.set('keyFetchToken', 'keyFetchToken');
+            account.set('unwrapBKey', 'unwrapBKey');
+            return broker.beforeSignUpConfirmationPoll(account);
+          })
+          .then(function () {
+            assert.equal(broker.session.oauth.keyFetchToken, 'keyFetchToken');
+            assert.equal(broker.session.oauth.unwrapBKey, 'unwrapBKey');
+          });
+      });
+    });
+
     describe('afterCompleteSignUp', function () {
-      it('calls sendOAuthResultToRelier', function () {
+      it('calls sendOAuthResultToRelier if there is session data present', function () {
+        setupCompletesOAuthTest();
+        return broker.persist()
+          .then(function () {
+            return broker.afterCompleteSignUp(account);
+          })
+          .then(function () {
+            assert.isTrue(broker.sendOAuthResultToRelier.called);
+            assert.isFalse(view.displayError.called);
+          });
+      });
+
+      it('doesnt call sendOAuthResultToRelier if there is no session data', function () {
```

Comment:
Heh, yeah I deliberately didn't do that because it looked funny in the JS; will add it.